platform.go

```diff
@@ -167,6 +167,21 @@ func NewBindingsFromPath(path string) (Bindings, error) {
 	return bindings, nil
 }
 
+// NewBindingsFromEnvOrPath creates a new bindings from all the bindings at the path defined by $SERVICE_BINDING_ROOT
+// or $CNB_BINDINGS if it does not exist.  If neither is defined, it defaults to using the given path.
+func NewBindingsFromEnvOrPath(path string) (Bindings, error) {
+	if path, ok := os.LookupEnv("SERVICE_BINDING_ROOT"); ok {
+		return NewBindingsFromPath(path)
+	}
+
+	// TODO: Remove as CNB_BINDINGS ages out
+	if path, ok := os.LookupEnv("CNB_BINDINGS"); ok {
+		return NewBindingsFromPath(path)
+	}
+
+	return NewBindingsFromPath(path)
+}
```

Comment:
```suggestion
// NewBindingsForBuild creates a new bindings from all the bindings at the path defined by $SERVICE_BINDING_ROOT
// or $CNB_BINDINGS if it does not exist.  If neither is defined, bindings are read from <platform>/bindings, the default
// path defined in the CNB Binding extension specification.
func NewBindingsForBuild(platformDir string) (Bindings, error) {
	if path, ok := os.LookupEnv("SERVICE_BINDING_ROOT"); ok {
		return NewBindingsFromPath(path)
	}

	// TODO: Remove as CNB_BINDINGS ages out
	if path, ok := os.LookupEnv("CNB_BINDINGS"); ok {
		return NewBindingsFromPath(path)
	}

	return NewBindingsFromPath(filepath.Join(platformDir, "bindings"))
}
```
I think we should do this ^ and rename `NewBindingsFromEnvironment` to `NewBindingsForLaunch` or similar to make it clear to users that the default path is only defined in the build environment. I also think it is nice to move the default path calculation into the helper function so all of the boilerplate for finding the bindings dir is in a single place.

Reply:
Renamed `NewBindingsFromEnvironment` but kept the old one around since it is still used.